src/lib/kafka_storage.ts

```diff
@@ -287,29 +306,41 @@ class Exporter {
       this.producer.produce(this.topicName, null, Buffer.from(event));
     });
 
-    return new Promise((resolve, reject) =>
-      this.producer.flush(KAFKA_FLUSH_TIMEOUT, (err, result) => {
+    return new Promise<void>((resolve, reject) =>
+      this.producer.flush(KAFKA_FLUSH_TIMEOUT, (err: LibrdKafkaError) => {
         if (err) return reject(err);
-        resolve(result);
+        resolve();
       })
     );
   }
 
-  async sendDataWithKey(events, keyField) {
+  async sendDataWithKey(events, keyField, signalRecordData = null) {
```

Comment:
There is the new logic in this PR. New argument here, to be sent to the partitions which do not get a payload.